.github/workflows/CI-python-vision.yml

```diff
@@ -0,0 +1,146 @@
+        packageDirectory:
+          ["erroranalysis", "rai_test_utils"]
+        operatingSystem: [ubuntu-latest, macos-latest, windows-latest]
+        pythonVersion: ["3.8", "3.9", "3.10"]
+        exclude:
+          - operatingSystem: macos-latest
+            pythonVersion: "3.8"
+
+    runs-on: ${{ matrix.operatingSystem }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.pythonVersion }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.pythonVersion }}
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: ${{ matrix.pythonVersion }}
+
+      - if: ${{ matrix.operatingSystem != 'macos-latest' }}
+        name: Install pytorch on non-MacOS
+        shell: bash -l {0}
+        run: |
+          conda install --yes --quiet pytorch torchvision captum cpuonly -c pytorch
+
+      - if: ${{ matrix.operatingSystem == 'macos-latest' }}
+        name: Install Anaconda packages on MacOS, which should not include cpuonly according to official docs
+        shell: bash -l {0}
+        run: |
+          conda install --yes --quiet pytorch torchvision captum -c pytorch
+
+      - if: ${{ matrix.operatingSystem == 'macos-latest' }}
+        name: Install latest lightgbm from conda-forge for MacOS
+        shell: bash -l {0}
+        run: |
+          conda install --yes --quiet lightgbm -c conda-forge
+
+      - name: Setup tools
+        shell: bash -l {0}
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade setuptools
+          pip install --upgrade "pip-tools<=7.1.0"
+
+      - name: Install package dependencies
+        shell: bash -l {0}
+        run: |
+          pip install -r requirements-dev.txt
+        working-directory: ${{ matrix.packageDirectory }}
+
+      - name: Install package extras
+        shell: bash -l {0}
+        run: |
+          pip install -r requirements-object-detection.txt
+        working-directory: ${{ matrix.packageDirectory }}
+
+      - name: Install package
+        shell: bash -l {0}
+        run: |
+          pip install -v -e .
+        working-directory: ${{ matrix.packageDirectory }}
+
+      - name: Pip freeze
+        shell: bash -l {0}
+        run: |
+          pip freeze > installed-requirements-dev.txt
+          cat installed-requirements-dev.txt
+        working-directory: raiwidgets
+
+      - name: Upload requirements
+        uses: actions/upload-artifact@v3
+        with:
+          name: requirements-dev.txt
+          path: raiwidgets/installed-requirements-dev.txt
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          pytest --durations=10 --doctest-modules --junitxml=junit/test-results.xml --cov=${{ matrix.packageDirectory }} --cov-report=xml --cov-report=html
+        working-directory: ${{ matrix.packageDirectory }}
+
+      - name: Upload code coverage results
```

Comment:
The code coverage is already uploaded as a part of CI-python.yml. Do we need to upload this report again? The coverage will anyways be very less.